packages/token/src/tests/erc20/test_erc20.cairo

```diff
@@ -449,3 +451,135 @@ fn test_burn_from_zero() {
     let mut state = setup();
     state.burn(ZERO(), VALUE);
 }
+
+//
+// update
+//
+
+#[test]
+fn test_update_from_non_zero_to_non_zero() {
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+
+    start_cheat_caller_address(contract_address, OWNER());
+
+    assert_state_before_transfer(OWNER(), RECIPIENT());
+    state.update(OWNER(), RECIPIENT(), VALUE);
+    assert_state_after_transfer(OWNER(), RECIPIENT(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, OWNER(), RECIPIENT(), VALUE);
+}
+
+#[test]
+fn test_update_from_non_zero_to_zero() {
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+
+    start_cheat_caller_address(contract_address, OWNER());
+
+    assert_state_before_burn(OWNER());
+    state.update(OWNER(), ZERO(), VALUE);
+    assert_state_after_burn(OWNER(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, OWNER(), ZERO(), VALUE);
+}
```

Comment:
Would be helpful to have a test covering an attempt to burn more than `total_supply`